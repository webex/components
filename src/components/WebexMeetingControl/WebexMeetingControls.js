import React from 'react';
import PropTypes from 'prop-types';

export const MeetingContext = React.createContext();

/**
 * WebexMeetingControls is a higher-order component that pass a meeting
 * context to several WebexMeetingControl components.
 *
 * @param {object} props
 * @returns {object} JSX of the component
 */
export default function WebexMeetingControls({meetingID, children}) {
  return (
    <MeetingContext.Provider value={meetingID}>
      <div className="meeting-controls">{children}</div>
    </MeetingContext.Provider>
  );
}

WebexMeetingControls.propTypes = {
  meetingID: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
