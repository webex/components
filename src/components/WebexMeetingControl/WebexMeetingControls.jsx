import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {MeetingContext} from '../hooks';
import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../../constants';

/**
 * WebexMeetingControls is a higher-order component that pass a meeting
 * context to several WebexMeetingControl components.
 *
 * @param {object} props
 * @returns {object} JSX of the component
 */
export default function WebexMeetingControls({children, className, meetingID}) {
  const mainClasses = {
    [`${WEBEX_COMPONENTS_CLASS_PREFIX}-meeting-controls`]: true,
    [className]: !!className,
  };

  return (
    <MeetingContext.Provider value={meetingID}>
      <div className={classNames(mainClasses)}>{children}</div>
    </MeetingContext.Provider>
  );
}

WebexMeetingControls.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  meetingID: PropTypes.string.isRequired,
};

WebexMeetingControls.defaultProps = {
  className: '',
};
