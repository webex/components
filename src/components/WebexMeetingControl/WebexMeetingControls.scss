.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-meeting-controls {
  display: flex;
  justify-content: center;
  align-items: center;

  > * {
    margin: 0.3125rem;
  }
}
