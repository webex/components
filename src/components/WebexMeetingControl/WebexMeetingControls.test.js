import React from 'react';
import TestRenderer from 'react-test-renderer';

import WebexMeetingControls from './WebexMeetingControls';

describe('Webex Meeting Controls', () => {
  describe('snapshot', () => {
    test('matches snapshot', () => {
      const component = TestRenderer.create(
        <WebexMeetingControls meetingID="my-meeting">
          <div id="test" />
        </WebexMeetingControls>,
      );

      expect(component.toJSON()).toMatchSnapshot();
    });

    test('matches snapshot with custom CSS class', () => {
      const component = TestRenderer.create(
        <WebexMeetingControls className="my-custom-css-class" meetingID="my-meeting">
          <div id="test" />
        </WebexMeetingControls>,
      );

      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
