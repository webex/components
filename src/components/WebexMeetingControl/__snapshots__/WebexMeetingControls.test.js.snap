// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webex Meeting Controls snapshot matches snapshot 1`] = `
<div
  className="wxc-meeting-controls"
>
  <div
    id="test"
  />
</div>
`;

exports[`Webex Meeting Controls snapshot matches snapshot with custom CSS class 1`] = `
<div
  className="wxc-meeting-controls my-custom-css-class"
>
  <div
    id="test"
  />
</div>
`;
